packages/stock-location/src/services/stock-location.ts

```diff
@@ -214,25 +233,26 @@ export default class StockLocationService {
     address: StockLocationAddressInput,
     context: { manager?: EntityManager } = {}
   ): Promise<StockLocationAddress> {
-    const manager = context.manager || this.getManager()
-    const locationAddressRepo = manager.getRepository(StockLocationAddress)
-
-    const existingAddress = await locationAddressRepo.findOne(addressId)
-    if (!existingAddress) {
-      throw new MedusaError(
-        MedusaError.Types.NOT_FOUND,
-        `StockLocation address with id ${addressId} was not found`
-      )
-    }
+    return await this.atomicPhase_(async (manager) => {
+      const locationAddressRepo = manager.getRepository(StockLocationAddress)
+
+      const existingAddress = await locationAddressRepo.findOne(addressId)
```

Comment:
thought: I believe `findOne` returns the first row in the table, if `undefined` is passed. Should we add 1) a check for defined `addressId`, as we do in the core, or 2) use a `where { id: addressId }` config, as this will work as expected?

Reply:
Good catch. undefined has to be validated even using where.
That also reminded me there are actually multiple places where `findOne` with optional relations should be replaced by `find` to prevent typeOrm from performing two queries